src/applications/static-pages/sharable-links/sharableLink.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
```

Comment:
I'm not familiar with this module. It looks like it allows us to define inline styles. Seems kinda cool, but also a bit of a departure from our current practice. Can you help me understand why you went this route?

Reply:
yup! I would imagine that the requirements for this ticket have introduced the most UI complexities to date, something that the outdated technology of scss files and vads helpers (including the passively aggressively named `shame.scss`) simply cannot handle. 

i went down that route and it led to very long classname strings, as well as lots of direct element manipulation (vs controlled components, which is a best practice) + adding and removing classes to classLists, and dynamic styling logic spread amongst a few different files - it was unmaintainable and pretty bug prone.

for example, one way to do the left alignment of the notification on mobile would have been with a [react portal](https://reactjs.org/docs/portals.html), but due to drupal, i couldn't, so I had to use css - no vads helpers could achieve dynamic alignment of a notification, and the only alternative was putting all my styles in `shame.scss`.

here's some more info:
https://styled-components.com/ 

please let me know if i answered/addressed the concerns, if not happy to jump on a zoom / call

Reply:
Thanks. I don't have a strong objection to this, but it does introduce yet another way of doing SCSS.

For the facilities static pages, we put our SCSS in `src/applications/static-pages/sass/modules/_m-facilities.scss`.

Reply:
Yeah the `shame.scss` file was named that way a long time ago because it was supposed to be designated for tech debt, or things that we intend to leave only temporarily. 

W/what @mpelzsherman said, there are SCSS files and modules other than that one. This component could probably have its own module at https://github.com/department-of-veterans-affairs/vets-website/tree/master/src/applications/static-pages/sass/modules.

I haven't used this myself so it might be great but since it's a divergence from SCSS and utility methods and we can't take advantage of centralized style variables (like `$color-primary` if you wanted to style something w/the main blue color), the platform will probably want to understand it before introducing it into the code base

Reply:
i don't see how we could apply styles in those `scss` files based on dynamic react variables. i think it's a pretty big technical limitation for complex UI interactions.  passing down a centralized style variable is definitely possible, i just didn't do it as a part of this PR. I will now, though ;)

https://styled-components.com/docs/advanced#theming